App/Composition/SmiliePicker/SmilieSearchViewModel.swift

```diff
@@ -0,0 +1,223 @@
+//  SmilieSearchViewModel.swift
+//
+//  Copyright 2025 Awful Contributors. CC BY-NC-SA 3.0 US https://github.com/Awful/Awful.app
+
+import Combine
+import CoreData
+import Foundation
+import Smilies
+
+@MainActor
+final class SmilieSearchViewModel: ObservableObject {
+    @Published var searchText = ""
+    @Published var allSmilies: [SmilieSection] = []
+    @Published var searchResults: [SmilieData] = []
+    @Published var recentlyUsedSmilies: [SmilieData] = []
+    @Published var isLoading = true
+    @Published var loadError: String?
+    
+    private let dataStore: SmilieDataStore
+    private var cancellables = Set<AnyCancellable>()
+    
+    struct SmilieSection {
+        let title: String
+        let smilies: [SmilieData]
+    }
+    
+    init(dataStore: SmilieDataStore) {
+        self.dataStore = dataStore
+        
+        setupSearchSubscription()
+        loadSmilies()
+    }
+    
+    private func setupSearchSubscription() {
+        $searchText
+            .debounce(for: .milliseconds(300), scheduler: DispatchQueue.main)
+            .removeDuplicates()
+            .sink { [weak self] searchText in
+                self?.performSearch(searchText)
+            }
+            .store(in: &cancellables)
+    }
+    
+    func loadSmilies() {
+        Task {
+            isLoading = true
+            loadError = nil
+            
+            await loadRecentlyUsed()
+            await loadAllSmilies()
+            
+            await MainActor.run {
```

Comment:
We're already on the main actor, so this seems redundant. If it's necessary, can you add a comment explaining why?

Reply:
I don't think it's necessary. Removing.